internal/translation/dbuser/conversion.go

```diff
@@ -0,0 +1,185 @@
+package dbuser
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/timeutil"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type User struct {
+	*akov2.AtlasDatabaseUserSpec
+	Password  string
+	ProjectID string
+}
+
+// NewUser wraps a Kubernetes Atlas User Spec pointer augmenting it with projectID and password.
+func NewUser(spec *akov2.AtlasDatabaseUserSpec, projectID, password string) *User {
+	return &User{AtlasDatabaseUserSpec: spec, ProjectID: projectID, Password: password}
+}
+
+// Normalize modifies the spec and returns it back. Clone to avoid unwanted changes.
+func Normalize(spec *akov2.AtlasDatabaseUserSpec) (*akov2.AtlasDatabaseUserSpec, error) {
```

Comment:
normalization is not complete. we have the following nested slices which need sorting:
- labels
- roles
- scopes

Reply:
let me fix that

Reply:
BTW note that ordering was already being done in the **fromAtlas** conversion direction. We can reuse that.

The fix will make `Normalize` an unexported implementation detail function which will be executed when data comes `fromAtlas` and when a `NewUser` is created (from the Kubernetes spec).

Also the diff code is moved into the translation layer conversion code, so it forces the user to use normalized internal user types.